### PR TITLE
Jj fix issues

### PR DESCRIPTION
**Pull request by JorMaister**
Stack of volumes is now imported without xmipp.
All wizards show the units in pixels and angtroms.

### CODE REVIEW

File: pwem/viewers/views.py [hunk 1]
```diff
@@ -21,17 +21,20 @@
...
-from pyworkflow.gui import getDefaultFont
-from pyworkflow.gui.text import openTextFileEditor
 import re
-import csv
 
+from pyworkflow.gui import getDefaultFont, openTextFileEditor
 
-import tkinter as tk
-import tkinter.ttk as ttk
+try:  # python 2
```
Comment (delarosatrevin) [inline pwem/viewers/views.py]: I think we should go fully with Python3 and remove Python2 support.

File: pwem/convert/headers.py [hunk 2]
```diff
@@ -230,8 +226,9 @@ def getCellDimensions(self):
     def getISPG(self):
         return self._header['ISPG']
 
-    def setISPG(self, ispg):
-        self._header['ISPG'] = ispg
+    def getNumberObjects(self):
```
Comment (delarosatrevin) [inline pwem/convert/headers.py]: getNumberOfObjects?

**Comment by fonsecareyna82** [comment]
@azazellochg, if you want we merge the branches right now

File: pwem/convert/headers.py [hunk 3]
```diff
@@ -336,13 +333,17 @@ def fixFile(cls, inFileName, outFileName, scipionOriginShifts,
...
-        if (ext == '.mrc') or (ext == '.map'):
-            return cls.MRC
-        elif (ext == '.spi') or (ext == '.vol'):
-            return cls.SPIDER
-        else:
-            return cls.UNKNOWNFORMAT
+def getFileFormat(fileName):
+
+    ext = getExt(cleanFileNameAnnotation(fileName))
+    if (ext == '.mrc') or (ext == '.map') or (ext == '.mrcs'):
```
Comment (delarosatrevin) [inline pwem/convert/headers.py]: parentheses are not required among cases

File: pwem/utils.py [hunk 4]
```diff
@@ -65,5 +65,10 @@ def getPWEMPath(*paths):
...
     return join(getPWEMPath('templates'), *paths)
 
+
 def getCmdPath(*paths):
-    return join(getPWEMPath('cmd'), *paths)
\ No newline at end of file
+    return join(getPWEMPath('cmd'), *paths)
+
+
+def convert_pix2length(sampling_rate, length):
```
Comment (delarosatrevin) [inline pwem/utils.py]: This function name does not follow our naming convention. ( camelCase)

File: pwem/utils.py [hunk 5]
```diff
@@ -65,5 +65,10 @@ def getPWEMPath(*paths):
...
+    return sampling_rate*length
```
Comment (delarosatrevin) [inline pwem/utils.py]: pep8?

File: pwem/wizards/wizard.py [hunk 6]
```diff
@@ -739,73 +751,242 @@ def _createPreview(self, frame):
...
+        self.preview = MaskPreview(frame, self.dim,
+                                   label=self.previewLabel,
+                                   listenersDict=listeners)
+
+        self.preview.grid(row=1, column=0)
+
+    def upKeyPress(self, event):
+        self.isInnerRad = False
+        self.highlightOuterSlider()
+
```
Comment (delarosatrevin) [inline pwem/wizards/wizard.py]: Too many empty lines

File: pwem/wizards/wizard.py [hunk 7]
```diff
@@ -739,73 +751,242 @@ def _createPreview(self, frame):
...
+        self.radiusSliderIn.grid(row=1, column=0, sticky='NEWS')
+        self.radiusSliderIn.columnconfigure(0, weight=1)
+        self.irLabel.grid(row=1, column=1, sticky='NSE', padx=5, pady=5)
+        self.irLabel.columnconfigure(0, weight=1)
+
+        self.manageMaskVals()
+
+    def updateSliderOuterRadius(self):
+
```
Comment (delarosatrevin) [inline pwem/wizards/wizard.py]: unnecessary empty line

File: pwem/wizards/wizard.py [hunk 8]
```diff
@@ -739,73 +751,242 @@ def _createPreview(self, frame):
...
+            self.isMakingBigger = False
+            # Check if the user is making the ring bigger
+            if new_val > self.outerRadius:
+                self.isMakingBigger = True
+
+            self.outerRadius = new_val
+            self.manageMaskVals()
+
+    def updateSliderInnerRadius(self):
+
```
Comment (delarosatrevin) [inline pwem/wizards/wizard.py]: unnecessary empty line

**Comment by azazellochg** [comment]
I'm away until next week, so go ahead if you want to

**Comment by pconesa** [comment]
thanks @azazellochg your suggestions were addressed, hopefully you'll like them.

**Pull request merged**